javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+/**
+ * This classloader is used to load arbitrary extensions for Otel Java instrumentation agent. Such
+ * extensions may include SDK components (exporters or propagators) and additional instrumentations.
+ * They have to be isolated and shaded to reduce interference with the user application and to make
+ * it compatible with shaded SDK used by the agent.
+ */
+// TODO find a way to initialize logging before using this class
+// TODO support scanning a folder for several extension jars and keep them isolated from each other
+// Used by AgentInitializer
+@SuppressWarnings("unused")
+public class ExtensionClassLoader extends URLClassLoader {
+  // NOTE it's important not to use slf4j in this class, because this class is used before slf4j is
+  // configured, and so using slf4j here would initialize slf4j-simple before we have a chance to
+  // configure the logging levels
```

Comment:
this restriction can be avoided by @laurit's comment https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2881/files#r628026661 (moving instantiation of ExtensionClassLoader inside of `AgentInstaller`)

Reply:
I want to address all changes around this when I will add support for multiple jars. I already see that will change instantiation logic.